src/expr/src/scalar/func.rs

```diff
@@ -3462,10 +3466,23 @@ fn split_part<'a>(datums: &[Datum<'a>]) -> Result<Datum<'a>, EvalError> {
     ))
 }
 
+fn is_ilike_pattern_match_dynamic<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    inner_is_like_pattern_match_dynamic(
+        a.unwrap_str().to_lowercase().as_str(),
+        b.unwrap_str().to_lowercase().as_str(),
+    )
+}
+
 fn is_like_pattern_match_dynamic<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
```

Comment:
Then you can just add a `case_insensitive: bool` here, and if true, call `needle.case_insensitive(case_insensitive)` before calling `needle.is_match`!

Reply:
I ended up adding flags to `like_pattern::build_regex` to avoid handling `.case_insensitive` in two places (`reduce` and `is_like_pattern_match_dynamic`), instead!

Reply:
Makes sense!